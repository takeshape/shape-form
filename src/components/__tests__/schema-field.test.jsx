import React from 'react';
import ReactShallowRenderer from 'react-test-renderer/shallow';
const renderer = new ReactShallowRenderer();

import SchemaField from '../schema-field';
import DefaultWidget from '../default-widget';

describe('SchemaField', () => {
  it('renders scalar field', () => {
    const props = {
      schema: {type: 'string'},
      config: {},
      formName: 'default',
      path: 'text'
    };

    const tree = renderer.render(<SchemaField {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it('renders array field', () => {
    const props = {
      schema: {type: 'array'},
      config: {},
      formName: 'default',
      path: 'text'
    };

    const tree = renderer.render(<SchemaField {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it('renders object field', () => {
    const props = {
      schema: {
        type: 'object',
        properties: {
          email: {type: 'string'},
          firstName: {type: 'string'},
          lastName: {type: 'string'}
        }
      },
      config: {},
      formName: 'default',
      path: 'user'
    };

    const tree = renderer.render(<SchemaField {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it('renders scalar field with custom widget', () => {
    const props = {
      schema: {type: 'string', title: 'Text Input'},
      config: {widget: 'text'},
      widgets: {
        text: DefaultWidget
      },
      formName: 'default',
      path: 'text'
    };

    const tree = renderer.render(<SchemaField {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
