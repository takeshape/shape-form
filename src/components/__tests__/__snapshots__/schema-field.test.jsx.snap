// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SchemaField renders array field 1`] = `
<Memo(Connect(ArrayField))
  config={Object {}}
  formName="default"
  path="text"
  schema={
    Object {
      "type": "array",
    }
  }
/>
`;

exports[`SchemaField renders object field 1`] = `
<ObjectField
  config={Object {}}
  formName="default"
  path="user"
  schema={
    Object {
      "properties": Object {
        "email": Object {
          "type": "string",
        },
        "firstName": Object {
          "type": "string",
        },
        "lastName": Object {
          "type": "string",
        },
      },
      "type": "object",
    }
  }
/>
`;

exports[`SchemaField renders scalar field 1`] = `
<Memo(Connect(ScalarField))
  config={Object {}}
  formName="default"
  path="text"
  schema={
    Object {
      "type": "string",
    }
  }
/>
`;

exports[`SchemaField renders scalar field with custom widget 1`] = `
<Memo(Connect(ScalarField))
  config={
    Object {
      "widget": "text",
    }
  }
  formName="default"
  path="text"
  schema={
    Object {
      "title": "Text Input",
      "type": "string",
    }
  }
  widgets={
    Object {
      "text": [Function],
    }
  }
/>
`;
